app/views/edit/build-config.html

```diff
@@ -536,7 +534,99 @@ <h3 class="with-divider">Run Policy
+                              <ui-select-match>{{$select.selected | sentenceCase}}</ui-select-match>
+                              <ui-select-choices repeat="type in buildHookTypes">
+                                {{type | sentenceCase }}
+                              </ui-select-choices>
+                            </ui-select>
+                          </div>
+
+                          <fieldset>
+                            <div ng-show="buildHookSelection.type === 'Shell Script'">
+                              <h4>Script</h4>
+                              <div
+                                ui-ace="{
+                                  mode: 'yaml',
+                                  theme: 'eclipse',
+                                  onLoad: aceLoaded,
+                                  rendererOptions: {
+                                    fadeFoldWidgets: true,
+                                    showPrintMargin: false
+                                  }
+                                }"
+                                ng-model="buildConfig.spec.postCommit.script"
+                                class="ace-bordered ace-read-only ace-inline dockerfile-mode mar-top-md">
+                              </div>
+                            </div>
+                            <div ng-show="buildHookSelection.type === 'Command'">
+                              <h4>Command</h4>
+                              <edit-command
+                                args="buildConfig.spec.postCommit.command"
+                                isRequired="true">
+                              </edit-command>
+                            </div>
+                            <div ng-show="buildHookSelection.type === 'Arguments to Default Image Entry Point'">
+                              <h4>Arguments to Default Image Entry Point</h4>
+                              <edit-command
+                                args="buildConfig.spec.postCommit.args"
+                                isRequired="true">
+                              </edit-command>
+                            </div>
+                            <div ng-show="buildHookSelection.type === 'Shell Script with Arguments'">
+                              <h4>Shell Script with Arguments</h4>
+                              <div
+                                ui-ace="{
+                                  mode: 'yaml',
+                                  theme: 'eclipse',
+                                  onLoad: aceLoaded,
+                                  rendererOptions: {
+                                    fadeFoldWidgets: true,
+                                    showPrintMargin: false
+                                  }
+                                }"
+                                ng-model="buildConfig.spec.postCommit.script"
+                                class="ace-bordered ace-read-only ace-inline dockerfile-mode mar-top-md">
```

Comment:
Same command for command and command with arguments. Let's not repeat the markup unless necessary.